plugins/InspectionGadgets/src/com/siyeh/ig/redundancy/RedundantStringOperationInspection.java

```diff
@@ -159,6 +165,80 @@ private ProblemDescriptor getStringConstructorProblem(PsiNewExpression expressio
+            return null;
+          }
+          return createProblem(equalsToCallExpression, anchor, RemoveRedundantChangeCaseFix.PlaceCaseEqualType.RIGHT);
+        }
+      }
+
+      PsiMethodCallExpression qualifierCall = MethodCallUtils.getQualifierMethodCall(call);
+      if (qualifierCall == null) return null;
+      PsiExpression receiver = qualifierCall.getMethodExpression().getQualifierExpression();
+      if (receiver == null) return null;
+
+      //cases:
+      //- text1.toLowerCase().equals("test2")
+      //- text1.toLowerCase().equals(text2.toLowerCase())
+      if (
+        isChangeCaseCall(qualifierCall)) {
+        PsiElement anchor = qualifierCall.getMethodExpression().getReferenceNameElement();
+        if (anchor == null) {
+          return null;
+        }
+        //case: text1.toLowerCase().equals("test2")
+        if (ExpressionUtils.computeConstantExpression(equalTo) instanceof String) {
+          return createProblem(qualifierCall, anchor, RemoveRedundantChangeCaseFix.PlaceCaseEqualType.LEFT);
+        }
+
+        //case: text1.toLowerCase().equals(text2.toLowerCase())
+        if (equalTo instanceof PsiMethodCallExpression) {
+          PsiMethodCallExpression secondCall = (PsiMethodCallExpression)equalTo;
+          if (isEqualChangeCaseCall(qualifierCall, secondCall)) {
+            return createProblem(secondCall, anchor, RemoveRedundantChangeCaseFix.PlaceCaseEqualType.BOTH);
+          }
+        }
+      }
+
+      return null;
+    }
+
+    @NotNull
+    private ProblemDescriptor createProblem(PsiMethodCallExpression equalsToCallExpression,
+                                            PsiElement anchor,
+                                            RemoveRedundantChangeCaseFix.PlaceCaseEqualType type) {
+      return myManager.createProblemDescriptor(anchor, (TextRange)null,
+                                               InspectionGadgetsBundle.message("inspection.redundant.string.call.may.be.message"),
+                                               ProblemHighlightType.LIKE_UNUSED_SYMBOL, myIsOnTheFly,
+                                               new RemoveRedundantChangeCaseFix(getChangeCaseType(equalsToCallExpression), type));
+    }
+
+
+    private static boolean isChangeCaseCall(@NotNull PsiMethodCallExpression qualifierCall) {
+      return (STRING_UPPER_CASE.test(qualifierCall) || STRING_LOWER_CASE.test(qualifierCall));
+    }
+
+    private static boolean isEqualChangeCaseCall(@NotNull PsiMethodCallExpression qualifierCall,
+                                                 @NotNull PsiMethodCallExpression secondCall) {
+      return (STRING_UPPER_CASE.test(secondCall) && STRING_UPPER_CASE.test(qualifierCall)) ||
+             (STRING_LOWER_CASE.test(secondCall) && STRING_LOWER_CASE.test(qualifierCall));
+    }
+
+    @NotNull
+    private static String getChangeCaseType(@NotNull PsiMethodCallExpression call) {
+      return STRING_UPPER_CASE.test(call) ? "toUpperCase()" : "toLowerCase()";
```

Comment:
Probably just `call.getMethodExpression().getReferenceName()`?

Reply:
Yes, thank you!